pgRouting 3.7.0 fails to compile under mingw64 gcc 13.2

I'm in the middle of switching my old mingw gcc 8.1 chain to my new mingw gcc 13.2 chain

**Problem**
While compiling and regressing works fine under my gcc 8.1 chain, it's failing at 59% of compile.
I've also tried pgRouting 3.6.2 and pgRouting 3.5.1, to rule out it was recent changes, and those fail as well.

This is compiling against PG17.  I haven't tried yet compiling with other versions of PostgreSQL.

Here is the output

> 
> [ 59%] Built target bellman_ford
> make  -f src/cpp_common/CMakeFiles/cpp_common.dir/build.make src/cpp_common/CMakeFiles/cpp_common.dir/depend
> make[2]: Entering directory '/projects/pgrouting/pgrouting-git/buildv3.7.0w64'
> C:/ming64/mingw64/bin/cmake.exe -E cmake_depends "Unix Makefiles" C:/ming64/projects/pgrouting/pgrouting-git C:/ming64/projects/pgrouting/pgrouting-git/src/cpp_common C:/ming64/projects/pgrouting/pgrouting-git/buildv3.7.0w64 C:/ming64/projects/pgrouting/pgrouting-git/buildv3.7.0w64/src/cpp_common C:/ming64/projects/pgrouting/pgrouting-git/buildv3.7.0w64/src/cpp_common/CMakeFiles/cpp_common.dir/DependInfo.cmake "--color="
> make[2]: Leaving directory '/projects/pgrouting/pgrouting-git/buildv3.7.0w64'
> make  -f src/cpp_common/CMakeFiles/cpp_common.dir/build.make src/cpp_common/CMakeFiles/cpp_common.dir/build
> make[2]: Entering directory '/projects/pgrouting/pgrouting-git/buildv3.7.0w64'
> [ 59%] Building CXX object src/cpp_common/CMakeFiles/cpp_common.dir/Dmatrix.cpp.obj
> cd C:/ming64/projects/pgrouting/pgrouting-git/buildv3.7.0w64/src/cpp_common && C:/ming64/mingw64/bin/c++.exe -DBOOST_ALLOW_DEPRECATED_HEADERS -DPGSQL_VERSION=17000 -DPROJECT_VERSION=\"3.7.0\" @CMakeFiles/cpp_common.dir/includes_CXX.rsp -frounding-math  -O3 -DNDEBUG -std=gnu++14   -Wsign-conversion -MD -MT src/cpp_common/CMakeFiles/cpp_common.dir/Dmatrix.cpp.obj -MF CMakeFiles/cpp_common.dir/Dmatrix.cpp.obj.d -o CMakeFiles/cpp_common.dir/Dmatrix.cpp.obj -c C:/ming64/projects/pgrouting/pgrouting-git/src/cpp_common/Dmatrix.cpp
> In file included from C:/ming64/projects/pgrouting/pgrouting-git/src/cpp_common/Dmatrix.cpp:25:
> C:/ming64/projects/pgrouting/pgrouting-git/include/cpp_common/Dmatrix.hpp:49:64: error: 'int64_t' was not declared in this scope
>    49 |     explicit Dmatrix(const std::map<std::pair<double, double>, int64_t> &euclidean_data);
>       |                                                                ^~~~~~~
> C:/ming64/projects/pgrouting/pgrouting-git/include/cpp_common/Dmatrix.hpp:37:1: note: 'int64_t' is defined in header '<cstdint>'; did you forget to '#include <cstdint>'?
>    36 | #include <utility>
>   +++ |+#include <cstdint>
>    37 |
> C:/ming64/projects/pgrouting/pgrouting-git/include/cpp_common/Dmatrix.hpp:49:71: error: template argument 2 is invalid
>    49 |     explicit Dmatrix(const std::map<std::pair<double, double>, int64_t> &euclidean_data);
>       |                                                                       ^
> C:/ming64/projects/pgrouting/pgrouting-git/include/cpp_common/Dmatrix.hpp:49:71: error: template argument 4 is invalid
> C:/ming64/projects/pgrouting/pgrouting-git/include/cpp_common/Dmatrix.hpp:70:17: error: 'int64_t' has not been declared
>    70 |     bool has_id(int64_t id) const;
>       |                 ^~~~~~~
> C:/ming64/projects/pgrouting/pgrouting-git/include/cpp_common/Dmatrix.hpp:77:22: error: 'int64_t' has not been declared
>    77 |     size_t get_index(int64_t id) const;
>       |                      ^~~~~~~
> C:/ming64/projects/pgrouting/pgrouting-git/include/cpp_common/Dmatrix.hpp:84:5: error: 'int64_t' does not name a type
>    84 |     int64_t get_id(size_t idx) const;
>       |     ^~~~~~~
> C:/ming64/projects/pgrouting/pgrouting-git/include/cpp_common/Dmatrix.hpp:84:5: note: 'int64_t' is defined in header '<cstdint>'; did you forget to '#include <cstdint>'?
> C:/ming64/projects/pgrouting/pgrouting-git/include/cpp_common/Dmatrix.hpp:104:21: error: 'int64_t' has not been declared
>   104 |     double distance(int64_t i, int64_t j) const {
>       |                     ^~~~~~~
> C:/ming64/projects/pgrouting/pgrouting-git/include/cpp_common/Dmatrix.hpp:104:32: error: 'int64_t' has not been declared
>   104 |     double distance(int64_t i, int64_t j) const {
>       |                                ^~~~~~~
> C:/ming64/projects/pgrouting/pgrouting-git/include/cpp_common/Dmatrix.hpp:120:17: error: 'int64_t' was not declared in this scope
>   120 |     std::vector<int64_t> ids;
>       |                 ^~~~~~~
> C:/ming64/projects/pgrouting/pgrouting-git/include/cpp_common/Dmatrix.hpp:120:17: note: 'int64_t' is defined in header '<cstdint>'; did you forget to '#include <cstdint>'?
> C:/ming64/projects/pgrouting/pgrouting-git/include/cpp_common/Dmatrix.hpp:120:24: error: template argument 1 is invalid
>   120 |     std::vector<int64_t> ids;
>       |                        ^
> C:/ming64/projects/pgrouting/pgrouting-git/include/cpp_common/Dmatrix.hpp:120:24: error: template argument 2 is invalid
> C:/ming64/projects/pgrouting/pgrouting-git/include/cpp_common/Dmatrix.hpp: In member function 'size_t pgrouting::tsp::Dmatrix::size() const':
> C:/ming64/projects/pgrouting/pgrouting-git/include/cpp_common/Dmatrix.hpp:90:37: error: request for member 'size' in '((const pgrouting::tsp::Dmatrix*)this)->pgrouting::tsp::Dmatrix::ids', which is of non-class type 'const int'
>    90 |     size_t size() const {return ids.size();}

--- a/NEWS.md
+++ b/NEWS.md
@@ -15,6 +15,8 @@ milestone for 3.7.1
 
 **Bug fixes**
 
+* [#2680](https://github.com/pgRouting/pgrouting/pull/2680) fails to compile
+  under mingw64 gcc 13.2
 * [#2689](https://github.com/pgRouting/pgrouting/pull/2689) When point is a
   vertex, the withPoints family do not return results.
 

--- a/doc/src/release_notes.rst
+++ b/doc/src/release_notes.rst
@@ -46,6 +46,8 @@ milestone for 3.7.1
 
 .. rubric:: Bug fixes
 
+* `#2680 <https://github.com/pgRouting/pgrouting/pull/2680>`__ fails to compile
+  under mingw64 gcc 13.2
 * `#2689 <https://github.com/pgRouting/pgrouting/pull/2689>`__ When point is a
   vertex, the withPoints family do not return results.
 

--- a/include/cpp_common/Dmatrix.hpp
+++ b/include/cpp_common/Dmatrix.hpp
@@ -1,8 +1,6 @@
 /*PGR-GNU*****************************************************************
 File: Dmatrix.hpp
 
-FILE: Dmatrix.h
-
 Copyright (c) 2015 pgRouting developers
 Mail: project@pgrouting.org
 
@@ -34,6 +32,7 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 #include <vector>
 #include <map>
 #include <utility>
+#include <cstdint>
 
 typedef struct IID_t_rt IID_t_rt;
 

